src/Preg.php

```diff
@@ -38,6 +38,28 @@ public static function match(string $pattern, string $subject, ?array &$matches
         return $result;
     }
 
+    /**
+     * @param non-empty-string   $pattern
+     * @param array<string> $matches Set by method
+     * @param int-mask<PREG_UNMATCHED_AS_NULL> $flags PREG_UNMATCHED_AS_NULL is always set, no other flags are supported
```

Comment:
If it is always set and no other flags are supported, why having that argument ?

Reply:
It's there for forward compatibility in case new flags are introduced which you may want to pass in (the phpdoc won't strictly allow them, but they're forwarded).

It's also there to keep BC with the preg_* signatures, so you can easily replace preg_ with Preg:: and migrate to this lib.